### PR TITLE
Make retry logic more permissive for network errors

### LINKED ISSUE #19235: Catch network errors in Gradle and retry with truncated exponential backoff
We should consider detecting network errors in gradle and retrying...

```dart
Initializing gradle...                                       6.3s
Resolving dependencies...                                       
* Error running Gradle:
Exit code 1 from: /home/travis/build/flutter/flutter sdk/examples/flutter_gallery/android/gradlew app:properties:
Starting a Gradle Daemon (subsequent builds will be faster)
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring root project 'android'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve com.android.tools.build:gradle:3.1.2.
     Required by:
         project :
      > Could not resolve com.android.tools.build:gradle:3.1.2.
         > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
            > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.1.2/gradle-3.1.2.pom'.
               > Remote host closed connection during handshake
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 24s
Please review your Gradle project setup in the android/ folder.
```

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/19235

The previous matcher string was too narrow - there are a number of ways that gradle can report failing to download in a manner that should be retriable, e.g. that `Remote host closed connection during handshake`, or that `connection timed out`. Rather than looking for a specific reason, just update the string to find the general problem and retry.

### CODE REVIEW

**Comment by dnfield** [comment]
One thing I'm not clear on - when we return the retry status, we don't do any backoff. That might work out ok because gradle takes long enough to get to the network failure anyway...?

File: packages/flutter_tools/lib/src/android/gradle_errors.dart [hunk 1]
```diff
@@ -118,7 +118,7 @@ final GradleHandledError networkErrorHandler = GradleHandledError(
     'javax.net.ssl.SSLHandshakeException: Remote host closed connection during handshake',
     'java.net.SocketException: Connection reset',
     'java.io.FileNotFoundException',
-    'Gateway Time-out'
```
Comment (blasten) [inline packages/flutter_tools/lib/src/android/gradle_errors.dart:121]: we need this one too, no?

Comment (dnfield) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:121]: If this one prints, we still get the new string I added to.

It doesn't hurt much to leave it - just means one more string to check through on a failure.

**Comment by blasten** [comment]
> One thing I'm not clear on - when we return the retry status, we don't do any backoff. 

We should backoff although it could just be constant.

File: packages/flutter_tools/lib/src/android/gradle_errors.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:121]: (See the test you added, which still passes, cause it contains the new string)

**Comment by dnfield** [comment]
Filed https://github.com/flutter/flutter/issues/83971

File: packages/flutter_tools/lib/src/android/gradle_errors.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:121]: 👍

**Review by blasten — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/android/gradle_errors.dart [hunk 2]
```diff
@@ -50,8 +50,8 @@ enum GradleBuildStatus {
   retryWithAarPlugins,
 }
 
-/// Returns a simple test function that evaluates to [true] if
-/// [errorMessage] is contained in the error message.
+/// Returns a simple test function that evaluates to `true` if at least one of
+/// `errorMessages` is contained in the error message.
```
Comment (blasten) [inline packages/flutter_tools/lib/src/android/gradle_errors.dart:54]: are we migrating to backticks?

Comment (dnfield) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:54]: No - but brackets mean dartdoc will try to link these to a class or class member. We don't use them for parameters or langauge keywords.

Comment (dnfield) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:54]: (I make this mistake all the time - now in the framework it gets caught but I don't think we check dartdoc validity in the tool)

Comment (blasten) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:54]: ah - I see

**Pull request merged**